scripts/eslint/rules/argument-trivia.cjs

```diff
@@ -0,0 +1,205 @@
+                        // requiring that arugments are tagged with the non-underscored name.
+                        return name.startsWith("_") ? name.slice(1) : name;
+                    }
+                }
+                return undefined;
+            });
+
+            const comments = sourceCode.getCommentsBefore(node);
+            /** @type {TSESTree.Comment | undefined} */
+            const comment = comments[comments.length - 1];
+
+            if (!comment || comment.type !== "Block") {
+                const expectedName = getExpectedName();
+                if (expectedName) {
+                    context.report({
+                        messageId: "argumentTriviaArgumentError",
+                        node,
+                        fix: (fixer) => {
+                            return fixer.insertTextBefore(node, `/*${expectedName}*/ `);
+                        }
+                    });
+                }
+                else {
+                    context.report({ messageId: "argumentTriviaArgumentError", node });
+                }
+                return;
+            }
+
+            const argRangeStart = node.range[0];
+            const commentRangeEnd = comment.range[1];
+            const expectedName = getExpectedName();
+            if (expectedName) {
+                const got = comment.value.trim();
+                if (got !== expectedName) {
+                    context.report({
+                        messageId: "argumentTriviaArgumentNameError",
+                        data: { got, want: expectedName },
+                        node: comment,
+                        fix: (fixer) => {
+                            // Try and preserve whitespace.
+                            const newComment = comment.value.replace(got, expectedName);
+                            return fixer.replaceText(comment, `/*${newComment}*/`);
+                        },
+                    });
+                    return;
+                }
+            }
+
+            const hasNewLine = sourceCodeText.slice(commentRangeEnd, argRangeStart).indexOf("\n") >= 0;
+            if (argRangeStart !== commentRangeEnd + 1 && !hasNewLine) {
+                // TODO(jakebailey): range should be whitespace
+                context.report({
+                    messageId: "argumentTriviaArgumentSpaceError",
+                    node,
+                    fix: (fixer) => {
+                        return fixer.replaceTextRange([commentRangeEnd, argRangeStart], " ");
+                    }
+                });
+            }
+        };
+
+        /** @type {(node: TSESTree.CallExpression | TSESTree.NewExpression) => void} */
+        const checkArgumentTrivia = (node) => {
+            if (shouldIgnoreCalledExpression(node)) {
+                return;
+            }
+
+            const getSignature = memoize(() => {
+                if (context.parserServices?.hasFullTypeInformation) {

```

Comment:
Huh, do you not always lint with full type information? Is there context around here? Just curious 😄

Reply:
Also, I've actually been trying to _drop_ type info, because it's just really expensive, but, we get too much value out of the singular type-using lint we have enabled, the one about unused type assertions.